zerver/data_import/slack_message_conversion.py

```diff
@@ -35,25 +48,25 @@
 # Hence, ~stri~ke doesn't format the word in Slack, but ~~stri~~ke
 # formats the word in Zulip
 SLACK_STRIKETHROUGH_REGEX = r"""
-                             (^|[ -(]|[+-/]|\*|\_|[:-?]|\{|\[|\||\^)     # Start after specified characters
-                             (\~)                                  # followed by an asterisk
-                                 ([ -)+-}—]*)([ -}]+)              # any character except asterisk
-                             (\~)                                  # followed by an asterisk
-                             ($|[ -']|[+-/]|[:-?]|\*|\_|\}|\)|\]|\||\^)  # ends with specified characters
+                             (\n|^|[ -(]|[+-/]|\*|\_|[:-?]|\{|\[|\||\^)     # Start after specified characters
+                             (\~)                                  # followed by a ~
+                                 ([^~]*)              # any character except ~
+                             (\~)                                  # followed by an ~
+                             (\n|$|[ -']|[+-/]|[:-?]|\*|\_|\}|\)|\]|\||\^)  # ends with specified characters
                              """
 SLACK_ITALIC_REGEX = r"""
-                      (^|[ -*]|[+-/]|[:-?]|\{|\[|\||\^|~)
+                      (\n|^|[ -*]|[+-/]|[:-?]|\{|\[|\||\^|~)
                       (\_)
-                          ([ -^`~—]*)([ -^`-~]+)                  # any character
+                          ([^_]*)                  # any character except _
                       (\_)
-                      ($|[ -']|[+-/]|[:-?]|\}|\)|\]|\*|\||\^|~)
+                      (\n|$|[ -']|[+-/]|[:-?]|\}|\)|\]|\*|\||\^|~)
                       """
 SLACK_BOLD_REGEX = r"""
-                    (^|[ -(]|[+-/]|[:-?]|\{|\[|\_|\||\^|~)
+                    (\n|^|[ -(]|[+-/]|[:-?]|\{|\[|\_|\||\^|~)
                     (\*)
-                        ([ -)+-~—]*)([ -)+-~]+)                   # any character
```

Comment:
The `([ -)+-~]+)` capture group is removed because is isn't used (`convert_markdown_syntax` doesn't use `match.group(5)). This applies to all other regexes

Reply:
CMIIW if this is important

Reply:
CMIIW and this turns out to be important

Reply:
This is not quite the right replacement.  The previous regex required at least one character within  the `*`s -- the new one does not.  This should be:
```suggestion
                        ([^*]+)                 # any character except *
```

My other comment is if you your commit message is a bulleted list of "key changes" then maybe each one should be its own commit. :)

Reply:
I did another look through on the two inner capture groups, and yeah still couldn't figure out why there are two of them... Confirmed that only using the second inner capture groups would suffice.

I added this commit [slack_regex: Remove unnecessary inner capture group.](https://github.com/zulip/zulip/pull/32164/commits/a254abdac1871d81f7acc0137f9844467064e1d9) to remove the first capture groups before we rework the inner capture group in the commit after that.